SimPEG/NSEM/FieldsNSEM.py

```diff
@@ -4,18 +4,133 @@
 from numpy.lib import recfunctions as recFunc
 from SimPEG.EM.Utils import omega
 
+
 ##############
 ### Fields ###
 ##############
-class BaseMTFields(Problem.Fields):
-    """Field Storage for a MT survey."""
+class BaseNSEMFields(Problem.Fields):
+    """Field Storage for a NSEM survey."""
     knownFields = {}
     dtype = complex
 
+###########
+# 1D Fields
+###########
+class Fields1D_ePrimSec(BaseNSEMFields):
+    """
+    Fields storage for the 1D NSEM solution.
+    """
+    knownFields = {'e_1dSolution':'F'}
+    aliasFields = {
+                    'e_1d' : ['e_1dSolution','F','_e'],
+                    'e_1dPrimary' : ['e_1dSolution','F','_ePrimary'],
+                    'e_1dSecondary' : ['e_1dSolution','F','_eSecondary'],
+                    'b_1d' : ['e_1dSolution','E','_b'],
+                    'b_1dPrimary' : ['e_1dSolution','E','_bPrimary'],
+                    'b_1dSecondary' : ['e_1dSolution','E','_bSecondary']
+                  }
+
+    def __init__(self,mesh,survey,**kwargs):
+        BaseNSEMFields.__init__(self,mesh,survey,**kwargs)
+
+    def _ePrimary(self, eSolution, srcList):
+        ePrimary = np.zeros_like(eSolution)
+        for i, src in enumerate(srcList):
+            ep = src.ePrimary(self.survey.prob)
+            if ep is not None:
+                ePrimary[:,i] = ep[:,-1]
+        return ePrimary
+
+    def _eSecondary(self, eSolution, srcList):
+        return eSolution
+
+    def _e(self, eSolution, srcList):
```

Comment:
could consider making this a part of an base NSEM fields object (should be consistent in 1D, 3D)